data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/CopyValueProcessor.java

```diff
@@ -13,47 +13,94 @@
 import org.opensearch.dataprepper.model.processor.AbstractProcessor;
 import org.opensearch.dataprepper.model.processor.Processor;
 import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 
 @DataPrepperPlugin(name = "copy_values", pluginType = Processor.class, pluginConfigurationType = CopyValueProcessorConfig.class)
 public class CopyValueProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private static final Logger LOG = LoggerFactory.getLogger(CopyValueProcessor.class);
+    private final CopyValueProcessorConfig config;
     private final List<CopyValueProcessorConfig.Entry> entries;
-
     private final ExpressionEvaluator expressionEvaluator;
 
     @DataPrepperPluginConstructor
     public CopyValueProcessor(final PluginMetrics pluginMetrics, final CopyValueProcessorConfig config, final ExpressionEvaluator expressionEvaluator) {
         super(pluginMetrics);
+        this.config = config;
         this.entries = config.getEntries();
         this.expressionEvaluator = expressionEvaluator;
     }
 
     @Override
     public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
         for(final Record<Event> record : records) {
-            final Event recordEvent = record.getData();
-            for(CopyValueProcessorConfig.Entry entry : entries) {
-                if (Objects.nonNull(entry.getCopyWhen()) && !expressionEvaluator.evaluateConditional(entry.getCopyWhen(), recordEvent)) {
-                    continue;
-                }
+            try {
+                final Event recordEvent = record.getData();
+                if (config.getFromList() != null || config.getToList() != null) {
```

Comment:
Is this check correct? What is the expected behavior if `getFromList()` returns null but `getToList()` is not null?

Reply:
I have a validation in `CopyValueProcessorConfig` that checks `from_list` and `to_list` should be both null or both non-null. So I assume this situation that `getFromList()` returns null but `getToList()` is not null won't happen.

Reply:
Regarding documentation, since we are adopting the new process to use the documentation website and remove readme (see #2740). I have this issue created to update over there: https://github.com/opensearch-project/documentation-website/issues/6390